packages/fast-usdc-contract/src/fast-usdc.flows.ts

```diff
@@ -132,3 +154,224 @@ export const forwardFunds = async (
+    if (!(accountId.namespace === 'cosmos' || supportsCctp(destination))) {
+      const destChain = chainOfAccount(destination);
+      statusManager.skipAdvance(evidence, [
+        `Transfer to ${destChain} not supported.`,
+      ]);
+      return;
+    }
+
+    const fullAmount = toAmount(evidence.tx.amount);
+    const { borrower, notifier, poolAccount } = config;
+    // do not advance if we've already received a mint/settlement
+    const mintedEarly = notifier.checkMintedEarly(evidence, destination);
+    if (mintedEarly) return;
+
+    // throws if requested does not exceed fees
+    const advanceAmount = feeTools.calculateAdvance(fullAmount, destination);
+
+    const tmpSeat = zcfTools.makeEmptyZCFSeat();
+    // throws if the pool has insufficient funds
+    borrower.borrow(tmpSeat, advanceAmount);
+
+    // this cannot throw since `.isSeen()` is called in the same turn
+    statusManager.advance(evidence);
+    const detail = {
+      txHash: evidence.txHash,
+      forwardingAddress: evidence.tx.forwardingAddress,
+      fullAmount,
+      destination,
+    };
+
+    try {
+      // XXX async flow types
+      await (localTransfer as unknown as GuestOf<typeof localTransfer>)(
+        tmpSeat,
+        poolAccount,
+        harden({ USDC: advanceAmount }),
+      );
+
+      // depositHandler.onfulFilled
+      tmpSeat.exit();
+      const amount = harden({
+        denom: usdc.denom,
+        value: advanceAmount.value,
+      });
+
+      const intermediateRecipient = getNobleICA().getAddress();
+
+      const destInfo = parseAccountIdArg(destination);
+      if (destInfo.namespace === 'cosmos') {
+        try {
+          await (destInfo.reference === settlementAddress.chainId
+            ? // local
+              poolAccount.send(destination, amount)
+            : // via IBC
+              poolAccount.transfer(destination, amount, {
+                forwardOpts: {
+                  intermediateRecipient,
+                },
+              }));
+          // transferHandler.onFulfilled
+          log('Advance succeeded', { advanceAmount, destination });
+          notifier.notifyAdvancingResult(detail, true);
+        } catch (error) {
+          await transferRejected(error);
+        }
+      } else if (supportsCctp(destination)) {
+        // send USDC via CCTP
+        try {
+          await poolAccount.transfer(intermediateRecipient, amount);
+          // transferCctpHandler.onFulfilled
+          // assets are on noble, transfer to dest.
+          const intermediaryAccount = getNobleICA();
+          try {
+            await intermediaryAccount.depositForBurn(destination, amount);
+            // transferHandler.onFulfilled
+            log('Advance succeeded', { advanceAmount, destination });
+            notifier.notifyAdvancingResult(detail, true);
+          } catch (error) {
+            await transferRejected(error);
+          }
+        } catch (error) {
+          transferCctpRejected(error);
+        }
```

Comment:
p.s. we lack coverage for both of these `catch` clauses.

Before the refactor, the lack over coverage was hidden by the fact that the `watch` for `depositForBurn` reused the `transferHandler`.

I'm somewhat inclined to flatten the `try`s and reduce their scope.